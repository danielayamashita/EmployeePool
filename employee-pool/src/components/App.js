import '../css/index.css';
import { connect} from "react-redux";
import {handleGetUsers,handleGetPools} from "../actions/shared"

import { Route, Routes } from "react-router-dom";

import {useEffect} from "react";

// import LoadingBar from "react-redux-loading-bar";

import LoginPage from './LoginPage';
import Dashboard from "./Dashboard"
import Leaderboard from "./Leaderboard"
import NewPool from "./NewPool"

const App = (props) => {

  useEffect(() => {
    props.dispatch(handleGetUsers());
    props.dispatch(handleGetPools());
  }, []);

  return (
    <div>
      {/* <LoadingBar /> */}
      <Routes>
        <Route path="/" exact element ={<LoginPage/>}/>
        <Route path="/Dashboard/:id" element ={<Dashboard/>}/>
        <Route path="/leaderboard/:id" element ={<Leaderboard/>}/>
        <Route path="/new/:id" element ={<NewPool/>}/>
      </Routes>
    </div>
  );
}

const mapStateToProps = ({  }) => ({
});

export default connect(mapStateToProps)(App);
