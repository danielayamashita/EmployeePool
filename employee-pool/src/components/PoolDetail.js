import { connect} from "react-redux";
import Nav from "./Nav";
import Option from "./Option"


const PoolDetail = (props) => {



    return (
        <div>
            <Nav/>
            <h3 className="center">{`Pool by ${props.detail.author}`}</h3>

            <h3 className="center">Would You Rather</h3>

            <div className="center">
                <Option type="optionOne"/>

                <Option type="optionTwo"/>
                
            </div>
        </div>
    );
}



const mapStateToProps = ({detailPool,pools,authedUser,users}) => ({
    users: users,
    authedUser: authedUser,
    detail: detailPool,
    allPools: pools
  });

export default connect(mapStateToProps)(PoolDetail);