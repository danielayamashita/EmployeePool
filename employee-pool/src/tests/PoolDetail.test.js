

import PoolDetail from "../components/PoolDetail";
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
// import { store } from './app/store';
import { MemoryRouter } from 'react-router'; 
import middleware from "../middleware"
import reducer from "../reducers";
import App from "../components/PoolDetail"
import { createStore } from "redux";


const store = createStore(reducer,middleware);

test('renders learn react link', () => {
  var component  = render(
    <MemoryRouter>
    <Provider store={store}>
      <PoolDetail />
    </Provider>
    </MemoryRouter>
  );

  expect(component).toMatchSnapshot();
});










