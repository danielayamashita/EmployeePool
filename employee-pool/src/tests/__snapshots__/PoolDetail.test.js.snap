// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            <p
              class="error-message"
            >
              404
            </p>
            <p
              class="error-submessage"
            >
               Question not found!
            </p>
            <p
              class="error-message"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M8.6 1c1.6.1 3.1.9 4.2 2 1.3 1.4 2 3.1 2 5.1 0 1.6-.6 3.1-1.6 4.4-1 1.2-2.4 2.1-4 2.4-1.6.3-3.2.1-4.6-.7-1.4-.8-2.5-2-3.1-3.5C.9 9.2.8 7.5 1.3 6c.5-1.6 1.4-2.9 2.8-3.8C5.4 1.3 7 .9 8.6 1zm.5 12.9c1.3-.3 2.5-1 3.4-2.1.8-1.1 1.3-2.4 1.2-3.8 0-1.6-.6-3.2-1.7-4.3-1-1-2.2-1.6-3.6-1.7-1.3-.1-2.7.2-3.8 1-1.1.8-1.9 1.9-2.3 3.3-.4 1.3-.4 2.7.2 4 .6 1.3 1.5 2.3 2.7 3 1.2.7 2.6.9 3.9.6zM7.9 7.5L10.3 5l.7.7-2.4 2.5 2.4 2.5-.7.7-2.4-2.5-2.4 2.5-.7-.7 2.4-2.5-2.4-2.5.7-.7 2.4 2.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          <p
            class="error-message"
          >
            404
          </p>
          <p
            class="error-submessage"
          >
             Question not found!
          </p>
          <p
            class="error-message"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.6 1c1.6.1 3.1.9 4.2 2 1.3 1.4 2 3.1 2 5.1 0 1.6-.6 3.1-1.6 4.4-1 1.2-2.4 2.1-4 2.4-1.6.3-3.2.1-4.6-.7-1.4-.8-2.5-2-3.1-3.5C.9 9.2.8 7.5 1.3 6c.5-1.6 1.4-2.9 2.8-3.8C5.4 1.3 7 .9 8.6 1zm.5 12.9c1.3-.3 2.5-1 3.4-2.1.8-1.1 1.3-2.4 1.2-3.8 0-1.6-.6-3.2-1.7-4.3-1-1-2.2-1.6-3.6-1.7-1.3-.1-2.7.2-3.8 1-1.1.8-1.9 1.9-2.3 3.3-.4 1.3-.4 2.7.2 4 .6 1.3 1.5 2.3 2.7 3 1.2.7 2.6.9 3.9.6zM7.9 7.5L10.3 5l.7.7-2.4 2.5 2.4 2.5-.7.7-2.4-2.5-2.4 2.5-.7-.7 2.4-2.5-2.4-2.5.7-.7 2.4 2.5z"
                fill-rule="evenodd"
              />
            </svg>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
